plugins/ipam/host-local/main.go

```diff
@@ -26,49 +28,99 @@ func main() {
 }
 
 func cmdAdd(args *skel.CmdArgs) error {
-	ipamConf, err := LoadIPAMConfig(args.StdinData, args.Args)
+	ipamConf, ipamConf6, err := LoadIPAMConfig(args.StdinData, args.Args)
 	if err != nil {
 		return err
 	}
 
-	store, err := disk.New(ipamConf.Name)
-	if err != nil {
-		return err
+	r := &types.Result{
+		IP4: nil,
+		IP6: nil,
 	}
-	defer store.Close()
 
-	allocator, err := NewIPAllocator(ipamConf, store)
-	if err != nil {
-		return err
-	}
+	if ipamConf != nil {
+		store, err := disk.New(ipamConf.Name)
+		if err != nil {
+			return err
+		}
+		defer store.Close()
 
-	ipConf, err := allocator.Get(args.ContainerID)
-	if err != nil {
-		return err
+		allocator, err := NewIPAllocator(ipamConf, store)
+		if err != nil {
+			return err
+		}
+
+		ipConf, err := allocator.Get(args.ContainerID)
+		if err != nil {
+			return err
+		}
+		r.IP4 = ipConf
 	}
 
-	r := &types.Result{
-		IP4: ipConf,
+	if ipamConf6 != nil {
```

Comment:
That's closer to what I had in mind, but will panic if there is no ipv4 ipamconf.
It's a question of style, but I tend to *first* acquire all handles, then perform the allocation. Something like

```go
var allocator4, allocator6 IPAllocator
var store4, store6 
if ipamConf4 {
    allocator4 = ...
    store4 = ...
}
if ipamConf6 {
    allocator6 = ...
    store6 = ...
}

if ipamConf4 != nil {
    ip4Conf, err := ip4allocator.Get ...
}
if ipamConf6 != nil {
    ip6Conf, err := ...
    if err != nil {
        if ipamConf4 != nil {
            allocator4.Release... 
        }
    }
}
```